core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
Yes, it means fail-fast, but only for locking :)

When it comes to RPCs, you already have 2 options to avoid blocking: `CACHE_MODE_LOCAL` or `FORCE_ASYNCHRONOUS`. (I see `putForExternalRead` uses the latter.) I don't think we need another option to block for RPCs, but only if there's no topology change...